src/libAtomVM/nifs.c

```diff
@@ -2469,3 +2494,281 @@ static term nif_console_print(Context *ctx, int argc, term argv[])
     }
     return OK_ATOM;
 }
+
+static char b64_table[64] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+// per https://tools.ietf.org/rfc/rfc4648.txt
+
+static term base64_encode(Context *ctx, int argc, term argv[], bool return_binary)
+{
+    UNUSED(argc);
+    term src = argv[0];
+
+    size_t src_size;
+    uint8_t *src_pos, *src_buf = NULL;
+    if (term_is_binary(src)) {
+        src_size = term_binary_size(src);
+        if (src_size == 0) {
+            return return_binary ? src : term_nil();
+        }
+        src_pos = src_buf;
+    } else if (term_is_list(src)) {
+        int ok;
+        src_size = interop_iolist_size(src, &ok);
```

Comment:
I checked again base64:(d)encode and it looks like it doesn't accept iolists.

Reply:
I say it is a "feature", and a good one at that!  Besides, a string is just a special case of an iolist.

Reply:
I also agree with you, we should document this addition.